### ISSUE TITLE
issue with tf.data.experimental.rejection_resample method

### ISSUE THREAD

**Issue by halhwadi** [tensorflow/tensorflow#47458]
when I run the below block of code i got this error:
"Shape must be rank 3 but is rank 2 for '{{node Tile}} = Tile[T=DT_FLOAT, Tmultiples=DT_INT32](ExpandDims, Tile/multiples)' with input shapes: [1,11,11], [2]."

I need to balance the "train_dataset" which has 11 labels by using the follwoing line of codes:

1) create class_fun:
```
def class_func(features, label):
  return label
```

2)create resampler object with targer_dist:
```
resampler = tf.data.experimental.rejection_resample(
    class_func, target_dist=[0.09, 0.09, 0.09, 0.09, 0.09, 0.09, 0.09, 0.09, 0.09, 0.09, 0.1])

```
When run the below line of code I got the error which I have mentioned it above :

`resample_ds = train_dataset.unbatch().apply(resampler)
`
Please advise?

**Comment by Saduf2019** [comment]
@halhwadi 
 I ran the code shared and face a different issue, please  find the [gist here](https://colab.research.google.com/gist/Saduf2019/e57b5fe550307126e26d8aea24221595/untitled554.ipynb). Please share all dependencies for us to replicate and analyse the issue faced.
